examples/agentworkflow/blog_writer.ts

```diff
@@ -21,47 +22,44 @@ const saveFileTool = FunctionTool.from(
     return `File saved successfully at ${filePath}`;
   },
   {
-    name: "saveFile",
-    description: "Save the provided content to a file",
-    parameters: {
-      type: "object",
-      properties: {
-        content: { type: "string" },
-      },
-      required: ["content"],
-    },
+    name: "save_file",
+    description:
+      "Save the written content into a file that can be downloaded by the user",
+    parameters: z.object({
+      content: z.string({
+        description: "The content to save into a file",
+      }),
+    }),
   },
 );
 
 async function main() {
-  const wikipediaTool = new WikipediaTool();
   const researchAgent = new FunctionAgent({
-    name: "research",
-    description: "A research agent that can search the web for information",
-    systemPrompt: `Always call the wikipedia tool to search the web for information first then handoff to other agent.`,
-    tools: [wikipediaTool],
-    canHandoffTo: ["report"],
+    name: "ResearchAgent",
+    description:
+      "Responsible for gathering relevant information from the internet",
+    systemPrompt: `You are a research agent. Your role is to gather information from the internet using the provided tools and then transfer this information to the writer agent for content creation.`,
+    tools: [new WikipediaTool()],
+    canHandoffTo: ["WriterAgent"],
```

Comment:
you can allow for `canHandoffTo` a `string[]` or `Agent[]`, then 
if you swap the order of the definitions of the agents, you can make this more typesafe like:
```suggestion
    canHandoffTo: reportAgent,
```